lib/validation.js

```diff
@@ -87,7 +87,17 @@ function compileSchemasForValidation (context, compile, isCustom) {
   }
 
   if (schema.body) {
-    context[bodySchema] = compile({ schema: schema.body, method, url, httpPart: 'body' })
+    const contentProperty = schema.body.content
+    if (contentProperty) {
+      const contentTypeSchemas = {}
+      for (const mediaName of Object.keys(contentProperty)) {
+        const contentSchema = contentProperty[mediaName].schema
+        contentTypeSchemas[mediaName] = compile({ schema: contentSchema, method, url, httpPart: 'body' })
```

Comment:
I would add the contentType. Right now it is not used, but a custom schema compiler may use it.

```suggestion
        contentTypeSchemas[mediaName] = compile({ schema: contentSchema, method, url, httpPart: 'body', contentType: mediaName })
```

Reply:
also improved naming consistency, the `mediaName` was used in the response PR so I aligned it with that but might wanna rename that to `contentType` everywhere for consistency